robin.user.js

```diff
@@ -72,30 +72,44 @@
 
     }
 
+    // Utils
+    function getChannelList()
+    {
+        var channels = String(settings.channel).split(",");
+        var channelArray = [];
+
+        for (i = 0; i < channels.length; i++)
+        {
+            var channel = channels[i].trim();
+            if (channel.length > 0)
+                channelArray.push(channel.toLowerCase());
+        }
 
+        return channelArray;
+    }
 
-
-    // Utils
-    function hasChannel(source, channel) {
-        channel = String(channel).toLowerCase().trim();
-        channel_array = channel.split(",");
+    function hasChannel(source)
+    {
+        channel_array = getChannelList();
         source = String(source).toLowerCase();
 
-        var idx = channel_array.length;
-        while(idx-- > 0) {
-            var current_chan = String(channel_array[idx]).toLowerCase().trim();
+        for (idx = 0; idx < channel_array.length; idx++)
+        {
```

Comment:
This hashChannel code was reverted from when I made the changes. I had it as a much more accurate function in my pull request here: https://github.com/daegalus/robin-grow/blob/master/robin.user.js#L66

Switching back will fix the ordering problem we have in the channel filter, and its more accurate.